### DISCUSSION TITLE
How to re-attach libraries after `LspRestart`?

### DISCUSSION THREAD

**Discussion by Davidyz** [folke/lazydev.nvim#56] [Q&A]
As the title says, I believe the lazyloaded libraries (`vim.*`) no longer works after I do a `LspRestart`. `nvim-cmp` no longer show the completion items and `lua_ls` raised warnings against all the `vim.*` symbols. Is there any way to automatically re-load the libraries whenever `lua_ls` get restarted?

**Comment by folke** [comment]
Why would you restart the lsp?
Either way, you can probably just do `:e` to fix it.

**Reply by Davidyz** [reply]
Hi thanks for your reply!

Sometimes certain lsp-related functionalities (completion, fold, etc.) doesn't behave properly and restarting lsp solves the issue (it might have something to do with how I configure the `lazy.nvim` plugin specs, but before I figure it out, restarting the lsp is a trivial fix). sometimes the language server simply freezes.
I tried `:e` and it still warns against `vim.*`.
